src/game/client/draw/drawable.h

```diff
@@ -82,6 +84,66 @@ class DrawableLocoInfo : public MemoryPoolObject
     TWheelInfo m_wheelInfo;
 };
 
+enum DrawableIconType
+{
+    ICON_HEAL,
+    ICON_HEAL_STRUCTURE,
+    ICON_HEAL_VEHICLE,
+    ICON_DEMORALIZED_OBSOLETE,
+    ICON_BOMB_TIMED,
+    ICON_BOMB_REMOTE,
+    ICON_DISABLED,
+    ICON_BATTLEPLAN_BOMBARDMENT,
+    ICON_BATTLEPLAN_HOLDTHELINE,
+    ICON_BATTLEPLAN_SEARCHANDDESTROY,
+    ICON_EMOTICON,
+    ICON_ENTHUSIASTIC,
+    ICON_SUBLIMINAL,
+    ICON_CARBOMB,
+    MAX_ICONS,
+    ICON_INVALID = -1,
+};
+
+class DrawableIconInfo : public MemoryPoolObject
+{
+    IMPLEMENT_POOL(DrawableIconInfo)
+
+public:
+    DrawableIconInfo()
+    {
+        for (int i = 0; i < MAX_ICONS; i++) {
+            anims[i] = nullptr;
+            timing[i] = 0;
+        }
+    }
+
+    void Kill_Icon(DrawableIconType icon)
+    {
+        if (anims[icon] != nullptr) {
+            anims[icon]->Delete_Instance();
+            anims[icon] = nullptr;
+            timing[icon] = 0;
+        }
+    }
+
+    void Reset()
+    {
+        for (int i = 0; i < MAX_ICONS; i++) {
+            if (anims[i] != nullptr) {
+                anims[i]->Delete_Instance();
+                anims[i] = nullptr;
+            }
+
+            timing[i] = 0;
+        }
+    }
+    virtual ~DrawableIconInfo() override { Reset(); }
+
+private:
+    Anim2D *anims[MAX_ICONS];
+    unsigned int timing[MAX_ICONS];
```

Comment:
Maybe word `timings` plural too because `anims` is plural.

Reply:
fixed.